actionpack/test/controller/request_forgery_protection_test.rb

```diff
@@ -706,6 +706,22 @@ def assert_blocked
       yield
     end
   end
+
+  def test_should_raise_for_an_origin_header_different_of_request_base_url
```

Comment:
Final tiny nitpick...
I think the test description should explain why we need this test. For example:
```suggestion
  def test_raised_exception_message_explains_why_it_occured
```